hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/S3EventsHoodieIncrSource.java

```diff
@@ -72,21 +73,23 @@ public S3EventsHoodieIncrSource(
       TypedProperties props,
       JavaSparkContext sparkContext,
       SparkSession sparkSession,
+      HoodieIngestionMetrics metrics,
```

Comment:
Based on the reflection, this argument should be after the `SchemaProvider` instance.

Reply:
Fixed.

Reply:
good catch